engines/e_pkcs11_eng.c

```diff
@@ -307,9 +323,121 @@ static EVP_PKEY *pkcs11_engine_load_private_key(ENGINE * e, const char *path,
     return 0;
 }
 
+static OSSL_STORE_LOADER_CTX* pkcs11_store_open(
+    const OSSL_STORE_LOADER *loader, const char *uri,
+    const UI_METHOD *ui_method, void *ui_data)
+{
+    size_t len;
+    ENGINE *e;
+    PKCS11_CTX *pkcs11_ctx;
+    OSSL_STORE_LOADER_CTX *store_ctx;
+
+    store_ctx = OSSL_STORE_LOADER_CTX_new();
+
+    len = strlen(pkcs11_scheme);
+    if (strncmp(uri, pkcs11_scheme, len) != 0)
+         return NULL;
```

Comment:
memory leak of `store_ctx ` and on each next `return NULL;`

Better declare : 
      `OSSL_STORE_LOADER_CTX *store_ctx = NULL;`  
and allocate/create it at last moment.